tasks/pkg-windows.yml

```diff
@@ -4,48 +4,81 @@
     msg: "The Datadog ansible role does not currently support Agent 5"
   when: datadog_agent_major_version|int == 5
 
+# This is a best-effort solution to avoid redownloading the msi when 
+# It won't work with rc / beta builds, and won't work if the version to install
+# is not pinned (as we don't know what version latest will be before downloading
+# the package).
+- name: Check currently installed Agent version
+  win_shell: |
+    $product_name = "Datadog Agent"
```

Comment:
On Chef we use `agent status` for this: https://github.com/DataDog/chef-datadog/blob/483f88a43569b1efd00e7d5c0cb1e479eb09b1a2/libraries/recipe_helpers.rb#L92

I'm not sure which approach is better, but I think we shouldn't have two.

Reply:
The Chef approach is slightly better because it would work with rc and beta builds. 

However, contrary to this method, it does have a caveat: it only works if you know where the Agent binary is located at (Chef expects it to be `C:/Program Files/Datadog/Datadog Agent/bin/agent`).

One thing to note is that we currently don't offer the possibility to choose the application directory for Windows install; we do have a `datadog_agent_binary_path_windows` parameter, but it's not advertised in the README (and thus I suppose it's an internal parameter) and only used for the integration command. We could use this option here to use `agent status`.

Reply:
If we have a variable for the Agent path, then I think using that to call the binary is reasonable (assuming we won't change the output of `agent status`).

I'm not sure this is necessarily better than your approach, though. I would still sync with @kbogtob, who wrote the Chef code I linked before, to discuss pros and cons of both approaches.